src/util/test_util.jl

```diff
@@ -0,0 +1,197 @@
+module TestUtils
+
+using AbstractGPs
+using Distributions
+using LinearAlgebra
+using Random
+using Test
+
+using AbstractGPs: AbstractGP, FiniteGP
+
+"""
+    test_finitegp_primary_public_interface(
+        rng::AbstractRNG, fx::FiniteGP; atol::Real=1e-12
+    )
+
+Basic consistency tests for the Primary Public FiniteGP API.
+You should run these tests if you only implement the Primary Public API -- see API section
+of docs for details about this API.
+
+These are consistency checks, not correctness tests in the absolute sense.
+For example, these tests ensure that samples generated by `rand` are of the correct size,
+but does not check that they come from the intended distribution.
+"""
+function test_finitegp_primary_public_interface(
+    rng::AbstractRNG, fx::FiniteGP; atol::Real=1e-12
+)
+    # Check that `rand` works at all, and produces something of the correct shape.
```

Comment:
```suggestion
    # Check that `rand` works, and produces something of the correct shape.
```
or alternatively `Check that 'rand' runs, ...`?